Scripts/Runtime/Entities/TaskDriver/AbstractTaskDriver.cs

```diff
@@ -0,0 +1,243 @@
+using Anvil.CSharp.Collections;
+using Anvil.CSharp.Core;
+using Anvil.CSharp.Logging;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Unity.Entities;
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    /// <summary>
+    /// Represents a context specific Task done via Jobs over a wide array of multiple instances of data.
+    /// The goal of a TaskDriver is to convert specific data into general data that the corresponding
+    /// <see cref="AbstractTaskDriverSystem"/> will process en mass and in parallel. The results of that general processing
+    /// are then picked up by the TaskDriver to be converted to specific data again and passed on to a sub task driver
+    /// or to another general system. 
+    /// </summary>
+    public abstract class AbstractTaskDriver : AbstractAnvilBase,
+                                               ITaskSetOwner
+    {
+        private static readonly Type TASK_DRIVER_SYSTEM_TYPE = typeof(TaskDriverSystem<>);
+
+        private readonly List<AbstractTaskDriver> m_SubTaskDrivers;
+        private readonly uint m_ID;
+        
+        private bool m_IsHardened;
+        private bool m_HasCancellableData;
+        
+        /// <summary>
+        /// Reference to the associated <see cref="World"/>
+        /// </summary>
+        public World World { get; }
```

Comment:
I don't mind making this change, but why? 

And why not the other getters?

Reply:
Cheaper no? It's not going `instance -> getter -> value` every time. just `instance -> value`.

I see now though that these getters are in place to satisfy the interface....so they need to stay as is.

Reply:
Ok yes! This is why I switched now. 

I typically do `private readonly` until something comes along where I need to make it `protected`, `public` or `internal`. For a while i was still doing readonly fields but then interfaces are what changed it and I just started always doing readonly getters so that I didn't have to change between fields and properties. 

You have another comment elsewhere where you looked it up and auto-getters get inlined so performance isn't an issue.